File: ejercicios-py/ejercicio10.py
import random
import string

"""
Genera un string de entre 16 y 40 caracteres aleatorios con letras y números.
:return: La contrasña generada
"""
def generate_string():
    length = random.randint(16, 40)
    password = ''
    
    chars = string.ascii_lowercase + string.digits
    
    while len(password) < length:
        password += random.choice(chars)
    
    return password[:length]

generated_str = generate_string()

print(f"La contraseña generada es: '{generated_str}' y tiene {len(generated_str)} caracteres")

File: ejercicios-js/ejercicio10.js
/**
 * Genera un string de entre 16 y 40 caracteres aleatorios con letras y números.
 * @result {str} La contrasña generada
 */
function generateString() {
    const length = Math.floor(Math.random() * 24) + 16;
    let password = '';

    while (password.length < length) {
        password += Math.random().toString(36).substring(2);
    }

    return password.slice(0, length);
}

const generatedStr = generateString();

console.log(`La contraseña generada es: '${generatedStr}' y tiene ${generatedStr.length} caracteres`);